ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerSubscription.scala

```diff
@@ -61,6 +65,21 @@ private[platform] case class ParallelIndexerSubscription[DB_BATCH](
               compressionStrategy = compressionStrategy,
             ),
             UpdateToMeteringDbDto(),
+            updatePackageMetadata = {
+              case packageUpload: state.Update.PublicPackageUpload =>
+                // new packages will be parsed and PackageMetadataView will be updated
+                // computation is executed on inputMapperExecutor
+                // since input-mapping precedes updating of the ledger-end, this ensures that upot ledger-end
+                // the PackageMetadataView will have all entries, but also means that updating is executed in
+                // a non-deterministic order, and PackageMetadataView will have more definitions and interface
+                // mappings.
+                // TODO DPP-1068: is the above approach okay? can it cause issues having inconsistently more entries?
```

Comment:
If I understand correctly, the issue is that when you query the `PackageMetadataView`, the result can include data from packages that do not exist at the current ledger end?

I don't see any issues with that in the current code. There might be a _very_ short window of time where you allow requests that refer to identifiers from not-yet-fully-indexed packages, but the filter transformation will still be correct because we're only adding implementations.

If we start removing implementations, then you could in theory have the situation where `PackageMetadataView` tells you not to use a template for the current offset range even though you should because it was only removed after the current `endInclusive`. I don't think this could happen in practice as you should only be able to remove packages that haven't been used in a long time.

Maybe you could document the guarantees on data liveness in `PackageMetadataView`, similar to https://github.com/digital-asset/daml/pull/14141/files#diff-8c6fbd5bcc3f222dfee7cb74bd6ffa355ab3f26edb3a1ec3ed4f10ae7ae06c3cR16-R23?